### ISSUE TITLE
[@typescript-eslint/ban-types] Generics that shadow a type are confused for those types

### ISSUE THREAD

**Issue by Vinnl** [typescript-eslint/typescript-eslint#2121]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": "error"
  }
}
```

```TS
interface ArbitraryParent {}
function someFunction<Object extends ArbitraryParent>(param: Object) {
  return param;
}
```

**Expected Result**
No problems.

**Actual Result**
```
error  Don't use 'Object' as a type. Use Record<string, any> instead
```

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.34.0` |
| `@typescript-eslint/parser`        | `2.34.0` |
| `TypeScript`                       | `3.9.3` |
| `ESLint`                           | `7.1.0` |
| `node`                             | `12.16.1` |
| `npm`                              | `6.13.4` |

**Comment by bradzacher** [comment]
The rule just looks at the name of the type. It doesn't know or care about if the type is defined locally or globally. This is done on purpose so that you can ban any types at all.

Shadowing global types like that is a definite code smell though, and you should avoid doing it as it makes your code a lot harder to understand.

**Comment by Vinnl** [comment]
OK, I presume it's a WONTFIX then.
